frontend/app/src/util/AppNavigation.ts

```diff
@@ -140,6 +149,128 @@ export class V1Strategy {
       ) ?? this.appPages[0]
     )
   }
+
+  clearPageElements(
+    _elements: AppRoot,
+    mainScriptHash: string,
+    sidebarElements: BlockNode | undefined
+  ): AppRoot {
+    return AppRoot.empty(mainScriptHash, false, sidebarElements)
+  }
+}
+
+export class V2Strategy {
```

Comment:
nit: I personally would call it `StrategyV2` amongst others as this might sort it better if there is a `Strategy` class